core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -330,6 +330,9 @@ private Object toJson(RexNode node) {
       final Object value = literal.getValue3();
       map = jsonBuilder.map();
       map.put("literal", value);
+      if (value instanceof Enum) {
+        map.put("flag", value.getClass().getName());
+      }
```

Comment:
Why we name the enum "flag" ? It makes me confused.

Reply:
Actually, I have considered using "type" or "enum". And RexBuilder has the method [makeFlag](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rex/RexBuilder.java#L876) to convert an Enum object to a RexLiteral, which will be used in `RelJson.toRex` in this patch. So choose  "flag" to be in consistent with it.

Reply:
So can we add this explain as comment of the code ?

Reply:
OK. That's good.